### DISCUSSION TITLE
Upload Speed

### DISCUSSION THREAD

**Discussion by david-steg** [nextcloud/all-in-one#609] [Questions] [answered]
Hi Everyone!
I was wondering, how fast is your Upload Speed by you guys?
It is clear that is depends on the type of file.
For a .Zip File, I reach up to 30MB/s, is it good or not? 
With a normal SMB Share I reach 100MB/s...

**Comment by szaimen** [comment, marked as answer]
Hi, you could try to point a local dns server like a pihole to the local ip of your Nextcloud. That will most likely improve the speed by a lot. Or did you already do that?

**Reply by david-steg** [reply]
![image](https://user-images.githubusercontent.com/71213394/167709019-527bc881-76a2-45de-8489-adc7febc3aff.png)
It's already setup. I unpluged the WAN, to be sure, same.

**Reply by ercoppa** [reply]
I am experiencing worse upload speeds with AIO compared to NCP (even when using a local DNS entry, which redirects to the local IP). I am using Nginx proxy manager as a reverse proxy (used also by NCP). I am getting slower upload speeds both in the web UI and in the desktop client. For instance, I am also seeing less than 30MB/s in the web UI, while with NCP I am getting ~50MB/s.

How can we debug this? Is anybody able to provide some upload speed benchmarks?

**Reply by fugohan** [reply]
I have the same problem but my whole network connections is getting routed via WAN. How can I change this behavior? If I try to login via the private IP I get redirected to the domain.
